code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -114,39 +125,119 @@
+
+	printing_amount = 0
+	for(var/datum/stock_part/servo/servo in component_parts)
+		printing_amount += servo.tier * 12.5
+	printing_amount = min(50, ROUND_UP(printing_amount))
+
+///Return a map of category->list of containers this machine can print
+/obj/machinery/chem_master/proc/load_printable_containers()
+	PROTECTED_PROC(TRUE)
+	SHOULD_BE_PURE(TRUE)
+
+	var/static/list/containers
+	if(!length(containers))
+		containers = list(
+			CAT_TUBES = GLOB.reagent_containers[CAT_TUBES],
+			CAT_PILLS = GLOB.reagent_containers[CAT_PILLS],
+			CAT_PATCHES = GLOB.reagent_containers[CAT_PATCHES],
+		)
+	return containers
+
+/obj/machinery/chem_master/item_interaction(mob/living/user, obj/item/tool, list/modifiers, is_right_clicking)
+	if(user.combat_mode || (tool.item_flags & ABSTRACT) || (tool.flags_1 & HOLOGRAM_1))
+		return ..()
+
+	if(is_reagent_container(tool) && tool.is_open_container())
+		replace_beaker(user, tool)
+		if(!panel_open)
+			ui_interact(user)
+			return ITEM_INTERACT_SUCCESS
+		else
+			return ITEM_INTERACT_BLOCKING
+
+	return ..()
+
 /obj/machinery/chem_master/wrench_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return NONE
+
+	. = ITEM_INTERACT_BLOCKING
+	if(is_printing)
+		balloon_alert(user, "still printing")
+		return
+
 	if(default_unfasten_wrench(user, tool) == SUCCESSFUL_UNFASTEN)
 		return ITEM_INTERACT_SUCCESS
-	return ITEM_INTERACT_BLOCKING
 
 /obj/machinery/chem_master/screwdriver_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return NONE
+
+	. = ITEM_INTERACT_BLOCKING
+	if(is_printing)
+		balloon_alert(user, "still printing")
+		return
+
 	if(default_deconstruction_screwdriver(user, icon_state, icon_state, tool))
-		update_appearance(UPDATE_ICON)
+		update_appearance(UPDATE_OVERLAYS)
 		return ITEM_INTERACT_SUCCESS
-	return ITEM_INTERACT_BLOCKING
 
 /obj/machinery/chem_master/crowbar_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return NONE
+
+	. = ITEM_INTERACT_BLOCKING
+	if(is_printing)
+		balloon_alert(user, "still printing")
+		return
+
 	if(default_deconstruction_crowbar(tool))
 		return ITEM_INTERACT_SUCCESS
-	return ITEM_INTERACT_BLOCKING
 
-/obj/machinery/chem_master/item_interaction(mob/living/user, obj/item/tool, list/modifiers, is_right_clicking)
-	if(is_reagent_container(tool) && !(tool.item_flags & ABSTRACT) && tool.is_open_container())
-		replace_beaker(user, tool)
-		if(!panel_open)
-			ui_interact(user)
-		return ITEM_INTERACT_SUCCESS
+/**
+ * Insert, remove, replace the existig beaker
+ * Arguments
+ *
+ * * mob/living/user - the player trying to replace the beaker
+ * * obj/item/reagent_containers/new_beaker - the beaker we are trying to insert, swap with existing or remove if null
+ */
+/obj/machinery/chem_master/proc/replace_beaker(mob/living/user, obj/item/reagent_containers/new_beaker)
+	PRIVATE_PROC(TRUE)
 
-	return ..()
+	if(!QDELETED(beaker))
+		try_put_in_hand(beaker, user)
+		beaker = null
```

Comment:
This `beaker = null` should actually be redundant given `/Exited` implementation
```suggestion
		try_put_in_hand(beaker, user)
```